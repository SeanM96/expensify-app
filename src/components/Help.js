import React from 'react';

const Help = () => (
    <div>
        This is from my add help component
    </div>
);

export default Help;