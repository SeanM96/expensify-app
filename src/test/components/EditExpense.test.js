import React from 'react';
import { shallow } from 'enzyme';
import { EditExpense } from '../../components/EditExpense';
import expenses from '../fixtures/expenses'

let editExpense; let removeExpense; let history; let wrapper;

beforeEach(() => {
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push: jest.fn() }
    wrapper = shallow(<EditExpense editExpense={editExpense} removeExpense={removeExpense} history={history} expense={expenses[1]}/>);
})

test('Edit expense page renders', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should handle onSubmit', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1]);
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(editExpense).toHaveBeenLastCalledWith(expenses[1].id, expenses[1]);
})

test('Should handle removeExpense', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenLastCalledWith('/');
    expect(removeExpense).toHaveBeenCalledWith({
        id: expenses[1].id
    });
})