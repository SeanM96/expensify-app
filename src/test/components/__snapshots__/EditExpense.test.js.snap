// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit expense page renders 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 200,
        "createdAt": -345600000,
        "description": "Test description two",
        "id": "2",
        "note": "Test Note two",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
    type="button"
  >
    Remove
  </button>
</div>
`;
